Detectors/PHOS/workflow/src/RawToCellConverterSpec.cxx

```diff
@@ -68,35 +59,42 @@ void RawToCellConverterSpec::init(framework::InitContext& ctx)
   if (fitmethod == "default") {
     LOG(INFO) << "Using default raw fitter";
     mRawFitter = std::unique_ptr<o2::phos::CaloRawFitter>(new o2::phos::CaloRawFitter);
-    //TODO: Configure parameters of fitter from options
-    // mRawFitter->setAmpCut(mNoiseThreshold);
-    // mRawFitter->setL1Phase(0.);
+  }
+  if (fitmethod == "semigaus") {
+    LOG(INFO) << "Using SemiGauss raw fitter";
+    mRawFitter = std::unique_ptr<o2::phos::CaloRawFitter>(new o2::phos::CaloRawFitterGS);
```

Comment:
mRawFitter is a pointer on the non-virtual CaloRawFitter class, which means that e.g. `evaluate` method of CaloRawFitter will be used, even if you cast the pointer from CaloRawFitterGS.
You should either use virtual methods or use templates.

Reply:
Dear @shahor02, this function will be called a lot. What would be more efficient: putting semiGS method to the base class CaloRawFitter and switch between methods with "if()" statement; Template or virtual member function?

Reply:
well, using a virtual methods has some small overhead, but given that the method itself is quite compute-intensive, I don't think you notice any difference.